# Econ 441A Assignment 5
## Juntao Li
## Student ID: 906092273

### 1/a)

In [1]:
import wooldridge as woo
import numpy as np
import statsmodels.formula.api as smf
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as stats
import linearmodels.iv as iv
mydata=woo.data('injury')

In [2]:
mydata

durat  afchnge  highearn  male  married  hosp  indust  injtype   age  \
0       1.0        1         1   1.0      0.0     1     3.0        1  26.0   
1       1.0        1         1   1.0      1.0     0     3.0        1  31.0   
2      84.0        1         1   1.0      1.0     1     3.0        1  37.0   
3       4.0        1         1   1.0      1.0     1     3.0        1  31.0   
4       1.0        1         1   1.0      1.0     0     3.0        1  23.0   
...     ...      ...       ...   ...      ...   ...     ...      ...   ...   
7145    1.0        0         0   1.0      1.0     0     3.0        7  23.0   
7146   18.0        0         0   0.0      0.0     0     3.0        7  31.0   
7147   14.0        0         0   0.0      0.0     0     3.0        7  66.0   
7148   76.0        0         0   0.0      NaN     0     3.0        7  39.0   
7149    4.0        0         0   1.0      1.0     0     3.0        7  45.0   

         prewage  ...  head  neck  upextr  trunk  lowback  lowextr  occdis  \
0     404.950012  ...     1     0       0      0        0        0       0   
1     643.825012  ...     1     0       0      0        0        0       0   
2     398.125000  ...     1     0       0      0        0        0       0   
3     527.799988  ...     1     0       0      0        0        0       0   
4     528.937500  ...     1     0       0      0        0        0       0   
...          ...  ...   ...   ...     ...    ...      ...      ...     ...   
7145  284.299988  ...     0     0       0      0        0        0       1   
7146  304.769592  ...     0     0       0      0        0        0       1   
7147  255.869995  ...     0     0       0      0        0        0       1   
7148  274.065186  ...     0     0       0      0        0        0       1   
7149  304.769592  ...     0     0       0      0        0        0       1   

      manuf  construc  highlpre  
0       0.0       0.0  6.003764  
1       0.0       0.0  6.467427  
2       0.0       0.0  5.986766  
3       0.0       0.0  6.268717  
4       0.0       0.0  6.270870  
...     ...       ...       ...  
7145    0.0       0.0  0.000000  
7146    0.0       0.0  0.000000  
7147    0.0       0.0  0.000000  
7148    0.0       0.0  0.000000  
7149    0.0       0.0  0.000000  

[7150 rows x 30 columns]

In [3]:
treat_model = smf.ols('ldurat ~ highearn + male + married + age + injtype', mydata).fit()
treat_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 ldurat   R-squared:                       0.025
Model:                            OLS   Adj. R-squared:                  0.024
Method:                 Least Squares   F-statistic:                     34.66
Date:                Mon, 28 Nov 2022   Prob (F-statistic):           4.06e-35
Time:                        15:22:02   Log-Likelihood:                -11434.
No. Observations:                6844   AIC:                         2.288e+04
Df Residuals:                    6838   BIC:                         2.292e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.7352      0.073     10.067      0.000       0.592       0.878
highearn       0.2544      0.036      7.122      0.000       0.184       0.324
male          -0.0331      0.041     -0.811      0.418      -0.113       0.047
married        0.0543      0.036      1.492      0.136      -0.017       0.126
age            0.0095      0.001      7.221      0.000       0.007       0.012
injtype        0.0351      0.010      3.422      0.001       0.015       0.055
==============================================================================
Omnibus:                       54.082   Durbin-Watson:                   1.899
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               83.069
Skew:                           0.039   Prob(JB):                     9.16e-19
Kurtosis:                       3.534   Cond. No.                         184.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The treatment effect on log(durat) is 0.2544 when control group is low-income workers, and the treatment group is high-income workers.

### 1/b)

In [4]:
test_model = smf.ols('highearn ~ male+married+age+injtype', mydata).fit()
test_model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               highearn   R-squared:                       0.205
Model:                            OLS   Adj. R-squared:                  0.205
Method:                 Least Squares   F-statistic:                     441.0
Date:                Mon, 28 Nov 2022   Prob (F-statistic):               0.00
Time:                        15:22:02   Log-Likelihood:                -4020.9
No. Observations:                6844   AIC:                             8052.
Df Residuals:                    6839   BIC:                             8086.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3063      0.024    -12.532      0.000      -0.354      -0.258
male           0.4083      0.013     31.668      0.000       0.383       0.434
married        0.2300      0.012     19.182      0.000       0.206       0.253
age            0.0052      0.000     11.800      0.000       0.004       0.006
injtype        0.0092      0.003      2.653      0.008       0.002       0.016
==============================================================================
Omnibus:                    13255.020   Durbin-Watson:                   0.336
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              481.965
Skew:                           0.139   Prob(JB):                    2.20e-105
Kurtosis:                       1.730   Cond. No.                         179.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The probability of F-stats is 0.00, which is extremely small and less than 0.05,so F-stats is significant. Thus the assignment to the treatment group is not random.

### 1/c)

In [5]:
test_model2 = smf.ols('ldurat ~ highearn*afchnge + male + married + age + injtype ', mydata).fit()
test_model2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 ldurat   R-squared:                       0.028
Model:                            OLS   Adj. R-squared:                  0.027
Method:                 Least Squares   F-statistic:                     28.01
Date:                Mon, 28 Nov 2022   Prob (F-statistic):           2.90e-38
Time:                        15:22:02   Log-Likelihood:                -11423.
No. Observations:                6844   AIC:                         2.286e+04
Df Residuals:                    6836   BIC:                         2.292e+04
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept            0.7315      0.075      9.716      0.000       0.584       0.879
highearn             0.1541      0.047      3.287      0.001       0.062       0.246
afchnge              0.0266      0.040      0.666      0.506      -0.052       0.105
highearn:afchnge     0.2049      0.064      3.217      0.001       0.080       0.330
male                -0.0338      0.041     -0.829      0.407      -0.114       0.046
married              0.0517      0.036      1.424      0.154      -0.019       0.123
age                  0.0094      0.001      7.188      0.000       0.007       0.012
injtype              0.0341      0.010      3.321      0.001       0.014       0.054
==============================================================================
Omnibus:                       52.250   Durbin-Watson:                   1.905
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               79.808
Skew:                           0.035   Prob(JB):                     4.68e-18
Kurtosis:                       3.524   Cond. No.                         200.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The treatment effect on log(durat) is 0.2049 by difference-in-difference analysis.

### 2/a)

In [6]:
mydata1=woo.data('FERTIL2')
mydata1

mnthborn  yearborn  age  electric  radio   tv  bicycle  educ  ceb  \
0            5        64   24       1.0    1.0  1.0      1.0    12    0   
1            1        56   32       1.0    1.0  1.0      1.0    13    3   
2            7        58   30       1.0    0.0  0.0      0.0     5    1   
3           11        45   42       1.0    0.0  1.0      0.0     4    3   
4            5        45   43       1.0    1.0  1.0      1.0    11    2   
...        ...       ...  ...       ...    ...  ...      ...   ...  ...   
4356         6        72   16       0.0    1.0  0.0      0.0     7    0   
4357         1        60   28       0.0    1.0  0.0      0.0     7    2   
4358        11        63   24       0.0    1.0  0.0      0.0     5    4   
4359         1        62   26       0.0    1.0  0.0      0.0     0    1   
4360        12        51   36       0.0    1.0  0.0      0.0     3    6   

      agefbrth  ...  heduc  agesq  urban  urb_educ  spirit  protest  catholic  \
0          NaN  ...    NaN    576      1        12       0        0         0   
1         25.0  ...   12.0   1024      1        13       0        0         0   
2         27.0  ...    7.0    900      1         5       1        0         0   
3         17.0  ...   11.0   1764      1         4       0        0         0   
4         24.0  ...   14.0   1849      1        11       0        1         0   
...        ...  ...    ...    ...    ...       ...     ...      ...       ...   
4356       NaN  ...    NaN    256      0         0       0        1         0   
4357      19.0  ...    NaN    784      0         0       0        1         0   
4358      13.0  ...    NaN    576      0         0       0        1         0   
4359      25.0  ...    7.0    676      0         0       1        0         0   
4360      22.0  ...    0.0   1296      0         0       0        1         0   

      frsthalf  educ0  evermarr  
0            1      0         0  
1            1      0         1  
2            0      0         1  
3            0      0         1  
4            1      0         1  
...        ...    ...       ...  
4356         1      0         0  
4357         1      0         0  
4358         0      0         0  
4359         1      1         1  
4360         0      0         1  

[4361 rows x 27 columns]

In [7]:
model1 = smf.ols('children~educ+age+I(age**2) ', mydata1).fit()
model1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               children   R-squared:                       0.569
Model:                            OLS   Adj. R-squared:                  0.568
Method:                 Least Squares   F-statistic:                     1915.
Date:                Mon, 28 Nov 2022   Prob (F-statistic):               0.00
Time:                        15:22:02   Log-Likelihood:                -7835.6
No. Observations:                4361   AIC:                         1.568e+04
Df Residuals:                    4357   BIC:                         1.570e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept      -4.1383      0.241    -17.200      0.000      -4.610      -3.667
educ           -0.0906      0.006    -15.298      0.000      -0.102      -0.079
age             0.3324      0.017     20.088      0.000       0.300       0.365
I(age ** 2)    -0.0026      0.000     -9.651      0.000      -0.003      -0.002
==============================================================================
Omnibus:                      203.406   Durbin-Watson:                   1.868
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              715.951
Skew:                           0.017   Prob(JB):                    3.41e-156
Kurtosis:                       4.985   Cond. No.                     1.07e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.07e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

From the regression estimates above, when age is fixed, the estimated effect of another year of education on fertility is -0.0906. If 100 women receive another year of education, they will expected to have 0.0906 * 100 fewer children, which is approximately 9 children.

### 2/b)

In [8]:
model2 = smf.ols('educ~frsthalf ', mydata1).fit()
model2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   educ   R-squared:                       0.014
Model:                            OLS   Adj. R-squared:                  0.014
Method:                 Least Squares   F-statistic:                     62.62
Date:                Mon, 28 Nov 2022   Prob (F-statistic):           3.15e-15
Time:                        15:22:02   Log-Likelihood:                -12122.
No. Observations:                4361   AIC:                         2.425e+04
Df Residuals:                    4359   BIC:                         2.426e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      6.3628      0.087     73.042      0.000       6.192       6.534
frsthalf      -0.9377      0.118     -7.913      0.000      -1.170      -0.705
==============================================================================
Omnibus:                       73.353   Durbin-Watson:                   1.396
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               41.939
Skew:                          -0.027   Prob(JB):                     7.82e-10
Kurtosis:                       2.523   Cond. No.                         2.72
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [9]:
hypothesis = ['frsthalf = 0']
model2.f_test(hypothesis)

<class 'statsmodels.stats.contrast.ContrastResults'>
<F test: F=62.62049531341785, p=3.1539715657891034e-15, df_denom=4.36e+03, df_num=1>

From the result of F-test above, we can see that F-test has p-value equal to 3.1539715657891034e-15, which is extremely small. This means F-statistic is significant in this model and it suggests that frsthalf is a strong IV.

### 2/c)

In [10]:
mydata1["educ_hat"]=model2.fittedvalues
model3 = smf.ols('children~educ_hat+age+I(age**2) ', mydata1).fit()
model3.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               children   R-squared:                       0.547
Model:                            OLS   Adj. R-squared:                  0.546
Method:                 Least Squares   F-statistic:                     1751.
Date:                Mon, 28 Nov 2022   Prob (F-statistic):               0.00
Time:                        15:22:02   Log-Likelihood:                -7944.5
No. Observations:                4361   AIC:                         1.590e+04
Df Residuals:                    4357   BIC:                         1.592e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept      -4.0583      0.374    -10.857      0.000      -4.791      -3.325
educ_hat       -0.1559      0.049     -3.212      0.001      -0.251      -0.061
age             0.3421      0.017     20.178      0.000       0.309       0.375
I(age ** 2)    -0.0026      0.000     -9.252      0.000      -0.003      -0.002
==============================================================================
Omnibus:                      199.972   Durbin-Watson:                   1.814
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              610.421
Skew:                           0.156   Prob(JB):                    2.81e-133
Kurtosis:                       4.806   Cond. No.                     1.63e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.63e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

The estimated effect of education here is -0.1559, which is less than -0.0906 in part a).

### 2/d)

In [11]:
model4 = smf.ols('children~educ+age+I(age**2)+electric+tv+bicycle ', mydata1).fit()
model4.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               children   R-squared:                       0.576
Model:                            OLS   Adj. R-squared:                  0.575
Method:                 Least Squares   F-statistic:                     984.9
Date:                Mon, 28 Nov 2022   Prob (F-statistic):               0.00
Time:                        15:22:02   Log-Likelihood:                -7789.3
No. Observations:                4356   AIC:                         1.559e+04
Df Residuals:                    4349   BIC:                         1.564e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept      -4.3898      0.240    -18.267      0.000      -4.861      -3.919
educ           -0.0767      0.006    -12.075      0.000      -0.089      -0.064
age             0.3402      0.016     20.692      0.000       0.308       0.372
I(age ** 2)    -0.0027      0.000    -10.010      0.000      -0.003      -0.002
electric       -0.3027      0.076     -3.974      0.000      -0.452      -0.153
tv             -0.2531      0.091     -2.768      0.006      -0.432      -0.074
bicycle         0.3179      0.049      6.440      0.000       0.221       0.415
==============================================================================
Omnibus:                      196.639   Durbin-Watson:                   1.891
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              675.502
Skew:                           0.009   Prob(JB):                    2.07e-147
Kurtosis:                       4.929   Cond. No.                     1.08e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.08e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [12]:
model5 = smf.ols('children~educ_hat+age+I(age**2)+electric+tv+bicycle ', mydata1).fit()
model5.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               children   R-squared:                       0.562
Model:                            OLS   Adj. R-squared:                  0.562
Method:                 Least Squares   F-statistic:                     931.9
Date:                Mon, 28 Nov 2022   Prob (F-statistic):               0.00
Time:                        15:22:02   Log-Likelihood:                -7858.0
No. Observations:                4356   AIC:                         1.573e+04
Df Residuals:                    4349   BIC:                         1.577e+04
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept      -4.3809      0.369    -11.884      0.000      -5.104      -3.658
educ_hat       -0.1203      0.048     -2.516      0.012      -0.214      -0.027
age             0.3505      0.017     21.016      0.000       0.318       0.383
I(age ** 2)    -0.0027      0.000     -9.807      0.000      -0.003      -0.002
electric       -0.4686      0.076     -6.161      0.000      -0.618      -0.319
tv             -0.4676      0.091     -5.136      0.000      -0.646      -0.289
bicycle         0.3037      0.050      6.056      0.000       0.205       0.402
==============================================================================
Omnibus:                      190.617   Durbin-Watson:                   1.872
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              591.855
Skew:                           0.119   Prob(JB):                    3.02e-129
Kurtosis:                       4.790   Cond. No.                     1.63e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.63e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

The estimated coefficient on educ in OLS is -0.0767, where the estimated coefficient on educ in 2SLS is -0.1203, so the coefficient in 2SLS is smaller. The coefficient of tv in OLS model is -0.2531, this means tv owners are estimated to have 0.2531 fewer children. The coefficient of tv in 2SLS model is -0.4676, this means tv owners are estimated to have 0.4676 fewer children. One of the reasons that television ownership has a negative effect on fertility is that tv owners may get information about birth control from tv.

### 3/a)

In [13]:
mydata2=woo.data('smoke')
mydata2

educ    cigpric  white  age  income  cigs  restaurn    lincome  agesq  \
0    16.0  60.506001      1   46   20000     0         0   9.903487   2116   
1    16.0  57.882999      1   40   30000     0         0  10.308952   1600   
2    12.0  57.664001      1   58   30000     3         0  10.308952   3364   
3    13.5  57.882999      1   30   20000     0         0   9.903487    900   
4    10.0  58.320000      1   17   20000     0         0   9.903487    289   
..    ...        ...    ...  ...     ...   ...       ...        ...    ...   
802  18.0  61.818001      0   52   30000    20         0  10.308952   2704   
803  18.0  61.675999      1   31   12500     0         0   9.433484    961   
804  16.0  60.707001      1   30   20000     0         0   9.903487    900   
805  10.0  59.987999      1   18   20000     0         0   9.903487    324   
806  10.0  59.652000      1   47   30000    20         0  10.308952   2209   

     lcigpric  
0    4.102743  
1    4.058424  
2    4.054633  
3    4.058424  
4    4.065945  
..        ...  
802  4.124195  
803  4.121895  
804  4.106059  
805  4.094144  
806  4.088528  

[807 rows x 10 columns]

I expect the sign of gamma_5 to be negative and the sign of gamma_6 to be also negative since higher cigarette prices and more restriction for smoking will both decrease the cigarette consumption.

### 3/b)

The assumption is that log(cigprice) and restaurn only changes cigs but not log(income). Then the income equation from part a) is identified.

### 3/c)

In [14]:
model6 = smf.ols('lincome~cigs+educ+age+I(age**2)', mydata2).fit()
model6.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                lincome   R-squared:                       0.165
Model:                            OLS   Adj. R-squared:                  0.161
Method:                 Least Squares   F-statistic:                     39.61
Date:                Mon, 28 Nov 2022   Prob (F-statistic):           2.68e-30
Time:                        15:22:03   Log-Likelihood:                -798.50
No. Observations:                 807   AIC:                             1607.
Df Residuals:                     802   BIC:                             1630.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept       7.7954      0.170     45.741      0.000       7.461       8.130
cigs            0.0017      0.002      1.010      0.313      -0.002       0.005
educ            0.0604      0.008      7.642      0.000       0.045       0.076
age             0.0577      0.008      7.548      0.000       0.043       0.073
I(age ** 2)    -0.0006   8.34e-05     -7.563      0.000      -0.001      -0.000
==============================================================================
Omnibus:                      264.025   Durbin-Watson:                   1.908
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              960.743
Skew:                          -1.531   Prob(JB):                    2.38e-209
Kurtosis:                       7.381   Cond. No.                     1.88e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.88e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

In [15]:
model7 = iv.IV2SLS.from_formula('lincome~1+educ+age+I(age**2)+[cigs ~ lcigpric+restaurn]',mydata2)
model7.fit(cov_type = 'unadjusted')

IV-2SLS Estimation Summary                          
==============================================================================
Dep. Variable:                lincome   R-squared:                     -0.5169
Estimator:                    IV-2SLS   Adj. R-squared:                -0.5245
No. Observations:                 807   F-statistic:                    89.802
Date:                Mon, Nov 28 2022   P-value (F-stat)                0.0000
Time:                        15:22:03   Distribution:                  chi2(4)
Cov. Estimator:            unadjusted                                         
                                                                              
                             Parameter Estimates                              
==============================================================================
            Parameter  Std. Err.     T-stat    P-value    Lower CI    Upper CI
------------------------------------------------------------------------------
Intercept      7.7809     0.2292     33.955     0.0000      7.3318      8.2300
I(age**2)     -0.0011     0.0003    -3.8424     0.0001     -0.0016     -0.0005
age            0.0938     0.0238     3.9454     0.0001      0.0472      0.1404
educ           0.0397     0.0162     2.4444     0.0145      0.0079      0.0715
cigs          -0.0421     0.0261    -1.6117     0.1070     -0.0934      0.0091
==============================================================================

Endogenous: cigs
Instruments: lcigpric, restaurn
Unadjusted Covariance (Homoskedastic)
Debiased: False
IVResults, id: 0x219c4278e80

The OLS estimate of beta_1 is 0.0017 while the 2SLS estimate of beta_1 is -0.0421. So the 2SLS estimate is smaller than the OLS estimate and it even changes the sign from positive from negative.

### 3/d)

I think they are exogenous in the income equation since they are instrumental variables of endogenous variable cigs and this means they are uncorrelated with the error term. So they are exogenous.